go/pinentry/pinentry_osx.go

```diff
@@ -78,11 +78,21 @@ func (pi *pinentryInstance) shouldStoreSecret(info pinentrySecretStoreInfo) bool
 	// up saying that the client wants to access the user's
 	// keychain. But this will do for now until we write our own
 	// pinentry.
-	attributes := kc.GenericPasswordAttributes{
-		ServiceName: pinentryServiceName,
-		AccountName: string(info),
+	query := keychain.NewItem()
+	query.SetSecClass(keychain.SecClassGenericPassword)
+	query.SetService(pinentryServiceName)
+	query.SetAccount(string(info))
+	query.SetMatchLimit(keychain.MatchLimitAll)
+	err := keychain.DeleteItem(query)
```

Comment:
We need that functionality, though. I think previously the keychain would prompt you for access before it deletes it.

Basically, the "save to keychain" checkbox in gpg pinentry saves the actual password to the keychain, which we don't want -- we just want a signal as to whether or not that checkbox was checked, and we can store that secret ourselves. So what we do is after pinentry is done, we check for the presence of a particular GPG entry (with the password) and use that as a signal, deleting it afterwards.

If we're going to nuke gpg pinentry support, we can just wait for that. Otherwise, we need an alternate way to do the same thing.